internal/api/server/server.go

```diff
@@ -0,0 +1,58 @@
+package server
+
+import (
+	"errors"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/edulinq/autograder/internal/api"
+	"github.com/edulinq/autograder/internal/common"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+// Run the autograder server and listen on an http and unix socket.
+func StartServer() (err error) {
```

Comment:
Make this something to indicate that we are not just starting and leaving, but waiting as well.
Maybe "RunServer" or "StartServerAndListen" like in http, or something like that.